modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/NestedMenus.ts

```diff
@@ -6,25 +6,83 @@ import { UiFactoryBackstage } from 'tinymce/themes/silver/backstage/Backstage';
 
 import ItemResponse from '../item/ItemResponse';
 import { expand } from './MenuConversion';
+import { MenuSearchMode } from './searchable/SearchableMenu';
 import { createPartialMenu } from './SingleMenu';
 import { SingleMenuItemSpec } from './SingleMenuTypes';
 
-// TODO: Consider moving the expansion part to alloy?
-const build = (items: string | Array<string | SingleMenuItemSpec>, itemResponse: ItemResponse, backstage: UiFactoryBackstage, isHorizontalMenu: boolean): Optional<TieredData> => {
+export interface NestedMenusSettings {
+  readonly isHorizontalMenu: boolean;
+  readonly search: Optional<{
+    readonly placeholder: Optional<string>;
+  }>;
+}
+
+const getSearchModeForField = (settings: NestedMenusSettings): MenuSearchMode => {
+  return settings.search.fold<MenuSearchMode>(
+    () => ({ searchMode: 'no-search' }),
+    (searchSettings) => {
+      return {
+        searchMode: 'search-with-field',
+        placeholder: searchSettings.placeholder
+      };
+    }
```

Comment:
```suggestion
    (searchSettings) => ({
      searchMode: 'search-with-field',
      placeholder: searchSettings.placeholder
    })
```
Up to you if you do this... I just figured it was more consistent given you used the shorthand version on the line above.